src/components/scenes/existingAccout/ChangeRecoveryScene.tsx

```diff
@@ -1,523 +1,439 @@
+          cancel: { label: s.strings.cancel, type: 'secondary' }
+        }}
+      >
+        {renderSaveRecoveryMessage(false)}
+      </ButtonsModal>
+    )).then(async button => {
+      if (button !== 'ok') return
+      await saveRecovery()
+    })
   }
 
-  renderItems = (item: { item: EdgeRecoveryQuestionChoice }) => {
-    return (
-      <TextRowComponent
-        data={item.item}
-        title={item.item.question}
-        onPress={this.handleQuestionSelected}
-        numberOfLines={3}
-      />
-    )
+  const changeRecovery = async () => {
+    if (account == null) return
+    Keyboard.dismiss()
+    try {
+      Airship.show(bridge => (
+        <ButtonsModal
+          bridge={bridge}
+          title={s.strings.recovery_change_title}
+          message={s.strings.recovery_change_message}
+          buttons={{
+            ok: {
+              label: s.strings.ok
+            },
+            cancel: {
+              label: s.strings.cancel,
+              type: 'secondary'
+            }
+          }}
+        />
+      )).then(async button => {
+        if (button !== 'ok') return
+        await clearAnswers().then(() => {
+          setRecoveryLocked(false)
+        })
+      })
+    } catch (error) {
+      showError(error)
+    }
   }
 
-  getQuestions = () => {
-    const { questionsList } = this.props
-    const { focusFirst, focusSecond, question1, question2 } = this.state
-    const questionFilter = focusFirst ? question2 : focusSecond ? question1 : ''
-    return questionsList.filter(item => item.question !== questionFilter)
+  const deleteRecovery = async () => {
+    if (account == null) return
+    Keyboard.dismiss()
+    try {
+      Airship.show(bridge => (
+        <ButtonsModal
+          bridge={bridge}
+          title={s.strings.recovery_reset_confirm_title}
+          message={s.strings.recovery_reset_confirm_message}
+          buttons={{
+            confirm: { label: s.strings.confirm },
+            cancel: { label: s.strings.cancel, type: 'secondary' }
+          }}
+        />
+      )).then(async button => {
+        console.log(button, s.strings.confirm, button === s.strings.confirm)
+        if (button !== 'confirm') return
+        await account.deleteRecovery()
+        await clearForm()
+        await done()
+      })
+    } catch (error) {
+      showError(error)
+    }
   }
 
-  renderQuestions = () => {
+  const renderQuestionAnswer = (index: number) => {
+    const question = extractQuestion(questions[index]) ?? questionPrompt
+    const answer = answers[index] ?? answerPrompt
     return (
-      <View style={styles.body}>
-        <FlatList
-          style={styles.questionsList}
-          data={this.getQuestions()}
-          renderItem={this.renderItems}
-          keyExtractor={(item, index) => index.toString()}
+      <View>
+        <Tile
+          type="touchable"
+          title={sprintf(s.strings.recovery_question, index + 1)}
+          body={question ?? questionPrompt}
```

Comment:
If showing the questionPrompt, style should change the text to a greyed out color: use `theme.deactivatedText`